pkg/ddl/ddl.go

```diff
@@ -1017,6 +1094,41 @@ func setDDLJobQuery(ctx sessionctx.Context, job *model.Job) {
 	}
 }
 
+func setDDLJobMode(job *model.Job) {
+	if variable.DDLVersion.Load() != model.TiDBDDLV2 {
+		job.LocalMode = false
+		return
+	}
+
+	switch job.Type {
+	// currently, v2 only support CreateTable without foreign keys.
+	case model.ActionCreateTable:
+		if len(job.Args) == 0 {
```

Comment:
When we will go to this logic? Or we can remove this logic.

Reply:
[8d1fd73](https://github.com/pingcap/tidb/pull/49687/commits/8d1fd73c70f8952a915ece922e82095e76905749)